generate_and_apply_patch.sh

```diff
@@ -6,11 +6,59 @@ BOOT=${1:-/boot}
 IRFS_HOOK="/etc/initramfs-tools/hooks/acpi_override.sh"
 cd `mktemp -d`
 
+###
+# If the tested version of iasl is not currently installed...
+if [ "$(iasl -v | grep -c 20181031)" -eq 0 ]; then
+ 
+ echo "[*] Attempting to install required tools to build iasl"
+ 
+ # Build a newer version of iasl
+ # If on Debian-based distro...
+ if [ -x "$(which apt-get)" ]; then
+     sudo apt-get -y install bison flex
+ # If on Arch-based distro...
+ elif [ -x "$(which pacman)" ]; then
+     sudo pacman -S bison flex
+ else
+     echo "Unable to automatically download and install bison or flex"
+     echo "Manual install needed"
+     exit 1
+ fi
+
+ echo "[*] Upgrading IASL..."
+ wget https://acpica.org/sites/acpica/files/acpica-unix2-20181031.tar.gz 
+ tar -zxf acpica-unix2-20181031.tar.gz
+ cd acpica-unix2-20181031
+ make clean
+ make iasl || { echo 'ERROR: Failed to build IASL' ; exit 1; }
+ sudo make install || { echo 'ERROR: Failed to install new version of IASL' ; exit 1; }
+ cd ..
+fi
+###
+
+###
 # Make sure we have required tools on systems with apt
-if [ -x $(which apt-get) ]; then
-	echo "[*] Installing required tools"
-	sudo apt-get -y install acpica-tools cpio
+echo "[*] Attempting to install required tools"
+
+# If on Debian-based distro...
+if [ -x "$(which apt-get)" ]; then
+    sudo apt-get -y install cpio
+
+# If on Arch-based distro...
+elif [ -x "$(which pacman)" ]; then
+    echo "[*] Installing Yay to be able to download AUR packages"
```

Comment:
Why do we need `yay` here. [cpio](https://www.archlinux.org/packages/extra/x86_64/cpio/) is normal package.

Reply:
Oh, I missed that one. I originally tried `pacman -S acpica cpio`, and when it failed, I assumed that both were not available in the official repositories. In that case, we don't need Yay at all then. We can just use:
```
pacman -Syu cpio
```